src/app/CMakeLists.txt

```diff
@@ -26,28 +26,42 @@
 #  POSSIBILITY OF SUCH DAMAGE.
 #
 
-add_library(commissioner-app
-    commissioner_app.cpp
-    json.cpp)
+add_library(commissioner-app)
 
-target_link_libraries(commissioner-app PUBLIC
-    commissioner
-    commissioner-common)
+target_sources(commissioner-app
+    PRIVATE
+        commissioner_app.cpp
+        commissioner_app.hpp
+        json.cpp
+        json.hpp
+)
 
-target_include_directories(commissioner-app PUBLIC
-    ${CMAKE_CURRENT_SOURCE_DIR})
+target_link_libraries(commissioner-app
+    PUBLIC
+        commissioner
+        commissioner-common
+)
 
-target_include_directories(commissioner-app PRIVATE
-    ${PROJECT_SOURCE_DIR}/third_party/json/repo/single_include)
+target_include_directories(commissioner-app
+    PUBLIC
+        ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
+target_include_directories(commissioner-app
+    PRIVATE
+        ${PROJECT_SOURCE_DIR}/third_party/json/repo/single_include
+)
```

Comment:
I'd think that if the target is added, e.g. `nlohman_json::nlohman_json`, then no need for doing this manually